security/fma-generic-contracts.md

```diff
@@ -0,0 +1,147 @@
+  Alternatively, we could detect it from monitoring or user reports that something seems off/incorrect.
+
+- **Recovery Path(s):**
+  Deploy new implementation with a fixed storage layout and write new values to storage.
+  Requires an upgrade.
+
+
+#### Proxy Initialized to Wrong Values
+
+- **Description:**
+    The proxy is initialized with the wrong values.
+
+- **Risk Assessment:**
+  High severity since arbitrary values can be incorrectly used.
+  Low likelihood since proxy initialization has a small surface area and is well verified/tested.
+
+- **Mitigations:**
+  Validate initialized values through extensive unit testing.
+  Simulations prior to performing the upgrade, verifying the initialized values.
+
+- **Detection:**
+  A simulation can show an unintended initialized proxy value.
+  Monitoring or user reports.
+
+- **Recovery Path(s):**
+  Repair the upgrade path (if needed).
+  Upgrade the smart contract.
+
+
+#### Can’t Upgrade Implementation
+
+- **Description:**
+  Contracts are unable to be upgraded.
+
+- **Risk Assessment:**
+  High severity since it effectively bricks an arbitrary contract in the system.
+  Low likelihood since contract upgrades are verified and follow industry-wide, standardized patterns.
+
+- **Mitigations:**
+  Validate through extensive unit testing.
+  Perform simulations that demonstrate upgradability.
+
+- **Detection:**
+  Simulate a contract upgrade to check for failure.
+
+- **Recovery Path(s):**
+  Upgrade contracts and/or proxies.
+  Update `op-chain-ops` if needed.
+
+
+#### Able to Reinitialize Implementation
+
+- **Description:**
+  A third party is able to reinitialize the contract to arbitrary values.
+
+- **Risk Assessment:**
+  High severity since an arbitrary actor can perform an action with arbitrary values.
+  Low likelihood since authentication is a critical, tested and validated property of the system.
+
+- **Mitigations:**
+  Validate through extensive unit testing.
+  Fuzz testing with random values.
+  Invariant testing that authentication properties hold.
+
+- **Detection:**
+  Simulate an arbitrary reinitialization.
+
+- **Recovery Path(s):**
+  Perform an upgrade that prevents the reinitialization.
+
+
+#### Backwards-incompatible ABIs
+
+- **Description:**
+  Functionality expected to hold across an upgrade or update breaks compatibility with prior systems.
+
+- **Risk Assessment:**
+  Low severity since forward-facing systems replace backwards systems.
+  Low likelihood since ABI changes require intentional changes.
+
+- **Mitigations:**
+  Validate ABI diffs.
+
+- **Detection:**
+  Breaking tools.
+  User reports.
+
+- **Recovery Path(s):**
+  Update integration paths for backwards-compatibility.
+  Some integrations (e.g. immutable contracts) may not be fixable.
+  Perform an upgrade to fix breaking change.
+
+
+#### Invalid `DisputeGameFactory.setImplementation` execution
```

Comment:
I was wondering why `Invalid DisputeGameFactory.setImplementation` is in this doc? 
As, all the previous bullet points are **generic upgrade practices** for smart contracts?
This one is specific to only FDG. Is there any particular reason?

Reply:
I may have misunderstood @mds1 [here](https://discord.com/channels/1244729134312198194/1298669460768423987/1299118978160922655) but I thought this was the intended outcome.

Reply:
I don't think this is supposed to be in the doc, I think that from the other doc (specific fma for 1.4) he was saying to link to this doc in a section for generic upgrade failures

Reply:
Sorry, I can see why you thought that from my comment, that should have said to link to it from the stage 1.4 FMA. I agree with @Ethnical and @tynes that this item is not specific to all contract upgrades so it shouldn't be here. We do already have this failure mode described in https://github.com/ethereum-optimism/design-docs/blob/main/security/fma-generic-hardfork.md#invalid-disputegamefactorysetimplementation-execution, so we can link to that from the specific stage 1.4 FMA

Reply:
Understood, will remove this section then, and like to `fma-generic-hardfork` in the stage 1.4 fma doc. Thanks all.